addons/escoria-core/game/core-scripts/esc/esc_object_manager.gd

```diff
@@ -15,96 +15,288 @@ const RESERVED_OBJECTS = [
 	CAMERA
 ]
 
+# The array of registered objects (organized by room, incl. one "room" for 
+# reserved objects).
+#
+# Example structure:
+#
+#	[
+#		{
+#			is_reserved: true,
+#			room: "",
+#			room_instance_id: "",
+#			objects:
+#				{
+#					"_camera": camera
+#				},
+#		},
+#		{
+#			is_reserved: false,
+#			room_global_id: "<room_global_id>",
+#			room_instance_id: "<room_object_instance_id>",
+#			objects:
+#				{
+#					"obj1": val1,
+#					"obj2": val2
+#				}
+#		}
+#	]
+#
+# Note that the "is_reserved" entry cannot be altered or otherwise changed and
+# that it belongs to no specific room.
+var room_objects: Array = []
+
+# We also store the current room's ids for retrieving the right objects.
+var current_room_key: ESCRoomObjectsKey
+
+# To avoid having to look this up all the time, we hold a reference.
+var reserved_objects_container: ESCRoomObjects
+
+
+func _init() -> void:
+	reserved_objects_container = ESCRoomObjects.new()
+	reserved_objects_container.is_reserved = true
+	reserved_objects_container.objects = {}
+	room_objects.push_back(reserved_objects_container)
 
-# The hash of registered objects (the global id is the key)
-var objects: Dictionary = {}
+	current_room_key = ESCRoomObjectsKey.new()
 
 
-# Make active objects visible
+# Make active objects in current room visible
 func _process(_delta):
-	for object in objects:
-		if (object as ESCObject).node:
-			(object as ESCObject).node.visible = (object as ESCObject).active
+	for room in room_objects:
+		if room.is_reserved or _is_current_room(room):
+			for object in room.objects:
+				if (object as ESCObject).node:
+					(object as ESCObject).node.visible = (object as ESCObject).active
```

Comment:
I believe it should, but I also know that this code pre-dates this PR, so maybe a question for @StraToN. I'm inclined to believe it will work, though, if the "active" property is ever set to true in ESC.

Reply:
I just tested and I confirm that an ESCitem whose `visible` attribute was set to false in the Inspector, does remain invisible.

Reply:
This code was left unchanged from the previous version of the object manager. If this needs to change, @StraToN can advise.

Reply:
Ah okay. Then we'd need to change it in another pr if it's wrong.